Signal/src/ViewControllers/ContactsViewHelper.m

```diff
@@ -340,21 +351,49 @@ - (void)presentContactViewControllerForRecipientId:(NSString *)recipientId
     }
 
     CNContactViewController *_Nullable contactViewController;
-    if (signalAccount) {
-        CNContact *_Nullable cnContact = signalAccount.contact.cnContact;
-        if (cnContact) {
-            if (shouldEditImmediately) {
-                // Not actually a "new" contact, but this brings up the edit form rather than the "Read" form
-                // saving our users a tap in some cases when we already know they want to edit.
-                contactViewController = [CNContactViewController viewControllerForNewContact:cnContact];
-
-                // Default title is "New Contact". We could give a more descriptive title, but anything
-                // seems redundant - the context is sufficiently clear.
-                contactViewController.title = @"";
-            } else {
-                contactViewController = [CNContactViewController viewControllerForContact:cnContact];
+    CNContact *_Nullable cnContact = nil;
+    if (addToExistingCnContact) {
+        CNMutableContact *updatedContact = [addToExistingCnContact mutableCopy];
+        NSMutableArray<CNLabeledValue *> *phoneNumbers
+            = (updatedContact.phoneNumbers ? [updatedContact.phoneNumbers mutableCopy] : [NSMutableArray new]);
+        // Only add recipientId as a phone number for the existing contact
+        // if its not already present.
+        BOOL hasPhoneNumber = NO;
+        for (CNLabeledValue *existingPhoneNumber in phoneNumbers) {
+            CNPhoneNumber *phoneNumber = existingPhoneNumber.value;
+            if ([phoneNumber.stringValue isEqualToString:recipientId]) {
+                hasPhoneNumber = YES;
```

Comment:
Done. Yeah, this was mostly about future-proofing this code since it's foreseeable that we could eventually add a "add to existing contact" action somewhere else.